app/src/main/java/com/plaid/linksample/MainActivityResultContractActivityJava.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2020 Plaid Technologies, Inc. <support@plaid.com>
+ */
+
+package com.plaid.linksample;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.TextView;
+import android.widget.Toast;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.plaid.link.OpenPlaidLink;
+import com.plaid.link.Plaid;
+import com.plaid.link.configuration.LinkTokenConfiguration;
+import com.plaid.link.result.LinkExit;
+import com.plaid.link.result.LinkSuccess;
+import com.plaid.linksample.network.LinkTokenRequester;
+
+import kotlin.Unit;
+
+
+public class MainActivityResultContractActivityJava extends AppCompatActivity {
+
+  private TextView result;
+  private TextView tokenResult;
+
+  // Experimental API using ActivityResultContract for androidx.fragment:1.3.0+
+  private ActivityResultLauncher<LinkTokenConfiguration> openPlaidLink =
+      this.registerForActivityResult(
+          new OpenPlaidLink(),
+          linkResult -> {
+            if (linkResult instanceof LinkSuccess) {
```

Comment:
NIT: might improve readability & create more consistency between different sample if we extract these to `handleSuccess` and `handleExit` methods explicitly

Reply:
Same for other occurrences with the `ActivityResultContract` or `LinkResultHandler`